x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -272,15 +272,29 @@ export const StepDetailsForm: FC<StepDetailsFormProps> = React.memo(
     const retentionPolicyMaxAgeEmpty = retentionPolicyMaxAge === '';
     const isRetentionPolicyMaxAgeValid = retentionPolicyMaxAgeValidator(retentionPolicyMaxAge);
 
-    // Reset retention policy settings when the user disables the whole option
     useEffect(() => {
+      // Reset retention policy settings when the user disables the whole option
       if (!isRetentionPolicyEnabled) {
         setRetentionPolicyDateField(
-          isRetentionPolicyAvailable ? dataViewAvailableTimeFields[0] : ''
+          isRetentionPolicyAvailable && dataViewAvailableTimeFields.length > 0
+            ? dataViewAvailableTimeFields[0]
+            : ''
         );
         setRetentionPolicyMaxAge('');
       }
-    }, [isRetentionPolicyEnabled]);
+
+      // When retention policy is first enabled, pick a default option
+      if (
+        isRetentionPolicyEnabled &&
```

Comment:
Should the retention policy field be disabled if there are no date fields in the dest index?

<img width="376" alt="image" src="https://github.com/elastic/kibana/assets/7405507/b6fc7765-d8af-4fe4-9c20-96b72d297217">

Currently I am able to create a transform which has no date field in the dest index, and set a retention policy with a blank retention policy `time.field`, but it then fails after creation:

<img width="1057" alt="image" src="https://github.com/elastic/kibana/assets/7405507/65e6b398-bfcb-481e-8b69-5a1701fcdfcf">

Reply:
Sorry, found another issue in the creation wizard. If you set up the transform to use an agg which adds a date/time field to the transform, progress to the details section, configure a retention policy, but then go back to step 1 and remove the date/time agg, the retention policy is left inside the config. This leads to the transform being created with

```
  "retention_policy": {
    "time": {
      "field": "",
      "max_age": "7d"
    }
  }
```

which causes the transform to fail on run due to the invalid setting.